Assets/Scripts/NPCs/BasicNurse.cs

```diff
@@ -0,0 +1,122 @@
+using UnityEngine;
+
+namespace NPCs
+{
+    [RequireComponent(typeof(Chaser))]
+    [RequireComponent(typeof(Responder))]
+    [RequireComponent(typeof(Searcher))]
+    [RequireComponent(typeof(Patroller))]
+    [RequireComponent(typeof(Waiter))]
+    [RequireComponent(typeof(Animator))]
+    public class BasicNurse : BaseAi
+    {
+        private static readonly HashSet<string> States = new HashSet<string>()
+        {
+            "Chasing",
+            "Responding",
+            "Searching",
+            "Patrolling",
+        };
+
+        [HideInInspector] public Chaser chaser;
+        [HideInInspector] public Responder responder;
+        [HideInInspector] public Searcher searcher;
+        [HideInInspector] public Patroller patroller;
+        [HideInInspector] public Waiter waiter;
+
+        public int numSearches = 3;
+        private int curNumSearches = 0;
+        
+        private GameObject questionMark;
+        private Animator animator;
+
+        private new void Awake()
+        {
+            base.Awake();
+
+            chaser = Utils.GetRequiredComponent<Chaser>(this);
+            responder = Utils.GetRequiredComponent<Responder>(this);
+            searcher = Utils.GetRequiredComponent<Searcher>(this);
+            patroller = Utils.GetRequiredComponent<Patroller>(this);
+            waiter = Utils.GetRequiredComponent<Waiter>(this);
+            animator = Utils.GetRequiredComponentInChildren<Animator>(this);
+
+            questionMark = transform.Find("QuestionMark").gameObject;
+            
+            SetState(currentState);
+        }
+
+        public override void SetState(string newState)
+        {
+            switch (newState)
+            {
+                case "Searching":
+                    agent.isStopped = false;
+                    agent.speed = searcher.movementSpeed;
+                    questionMark.SetActive(true);
+                    break;
+                case "Responding":
+                    agent.isStopped = false;
+                    agent.speed = responder.movementSpeed;
+                    questionMark.SetActive(true);
+                    break;
+                case "Chasing":
+                    agent.isStopped = false;
+                    agent.speed = chaser.movementSpeed;
+                    questionMark.SetActive(false);
+                    break;
+                case "Patrolling":
+                    agent.isStopped = false;
+                    agent.speed = patroller.movementSpeed;
+                    questionMark.SetActive(false);
+                    break;
+                default:
+                    questionMark.SetActive(false);
+                    throw new UnityException("Invalid state name passed to " + GetType().Name);
+            }
+        
+            currentState = newState;
+        }
+
+        protected override void Update()
+        {
+            animator.SetBool(Constants.ANIMATION_STEVE_MOVING, !agent.isStopped);
```

Comment:
I should mention that we have a walking animation as well sitting in the prefabs - maybe walking when patrolling/searching and running when chasing?

Reply:
Sounds good, do you know what the name for that animation is in Constants? (e.g. Constants.ANIMATION_STEVE_WALKING ?)

Reply:
Walking isn't set up yet, so there aren't any animation states or logic or constants set up.
We can add walking in a separate PR, I can look into it after this is merged in since I did the running part of it.